homeassistant/helpers/script.py

```diff
@@ -1380,6 +1381,44 @@ def supports_max(self) -> bool:
         """Return true if the current mode support max."""
         return self.script_mode in (SCRIPT_MODE_PARALLEL, SCRIPT_MODE_QUEUED)
 
+    @cached_property
+    def referenced_floors(self) -> set[str]:
+        """Return a set of referenced fooors."""
+        referenced_floors: set[str] = set()
+        Script._find_referenced_floors(referenced_floors, self.sequence)
+        return referenced_floors
+
+    @staticmethod
+    def _find_referenced_floors(
```

Comment:
This function shares a lot of code with `_find_referenced_areas` and `_find_referenced_devices`.
Can we create one private function out of it and reduce the duplicated code?

Into the new common function, we need to pass in `ATTR_FLOOR_ID` and `Script._find_referenced_floors`

Reply:
I've merged the floors/areas finding.

I've not merge it with devices & entities at this point. While they are similar, they are different (e.g., handling of condition logic). I didn't want to put that refactor into the scope of this PR